### ISSUE TITLE
Add Typescript Types

### ISSUE THREAD

**Issue by pascalkania** [Cidaas/cidaas-javascript-sdk#14]
hey guys, 

to make the library compatible with typescript, it would be nice if the specific types were generated for it.

the following link contains a more detailed description of how to generate them with typescript.
https://www.typescriptlang.org/docs/handbook/declaration-files/dts-from-js.html
 
otherwise i can also do the work, please just let me know ;)

**Comment by sawi** [comment]
@pascalkania you are right, a deeper compatibility with typescript will be great and we are currently working on this. It's great that you want to support us with this, if you want I can connect you with our development team ;-)

**Comment by pascalkania** [comment]
thank you @sawi , but the types have already been added. 

I also have noticed that the library oidc-client-js is used internally. Since last year this library will not be maintained any more. 

As an alternative there is a fork which is also completely based on Typescript. You can find it under the following URL: https://github.com/authts/oidc-client-ts

i also found that there is a library for react which is also based on this fork, you can find it here: https://github.com/authts/react-oidc-context

I just wanted to inform you about it, in case you haven't heard about it yet.

**Comment by igorlino** [comment]
@sawi hi,  any updates on this .....
